storm-webapp/src/main/java/org/apache/storm/daemon/ui/UIHelpers.java

```diff
@@ -1488,16 +1495,21 @@ private static Double nullToZero(Double value) {
      * @return getTopologySpoutAggStatsMap
      */
     private static Map<String, Object> getTopologySpoutAggStatsMap(ComponentAggregateStats componentAggregateStats,
-                                                                   String spoutId) {
+                                                                   String spoutId, Map<String, Object> config, String topologyId) {
         Map<String, Object> result = new HashMap();
         CommonAggregateStats commonStats = componentAggregateStats.get_common_stats();
         result.putAll(getCommonAggStatsMap(commonStats));
         result.put("spoutId", spoutId);
         result.put("encodedSpoutId", Utils.urlEncodeUtf8(spoutId));
         SpoutAggregateStats spoutAggregateStats = componentAggregateStats.get_specific_stats().get_spout();
-        result.put("completeLatency", spoutAggregateStats.get_complete_latency_ms());
+        result.put("completeLatency", StatsUtil.floatStr(spoutAggregateStats.get_complete_latency_ms()));
         ErrorInfo lastError = componentAggregateStats.get_last_error();
-        result.put("lastError", Objects.isNull(lastError) ?  "" : getTruncatedErrorString(lastError.get_error()));
+        if (!Objects.isNull(lastError)) {
+            result.putAll(getComponentErrorInfo(lastError, config, topologyId, true));
+        } else {
+            // Maintain backwards compatibility in the API response by setting empty string value
```

Comment:
Since the previous behavior always defaulted `lastError` to empty string, and the API docs show it defaulting to empty string, I've left this here for backwards compatibility. 

If you think it'd be better to just exclude it (as we do with the other keys like `errorPort`, `errorHost`, `errorTime` etc.. let me know and I can remove the else block.

Reply:
You are right. Sorry I overlooked. I think we can default those keys to empty like you said

Reply:
Sorry, it's unclear, would you prefer that I also default `errorPort`, `errorHost`, and `errorTime` to empty string?

Thanks!

Reply:
Sorry for not being clear. I think `lastError`, `errorHost`, `errorPort`, `errorWorkerLogLink` can default to empty string, and `errorTime`, `errorLapsedSecs` can default to `null` . What do you think?

Reply:
Sounds good.  I'll update the code and API docs to reflect that.

Thanks!

Reply:
Updated the code as well as the API docs.

Thanks!